typescript/index.ts

```diff
@@ -0,0 +1,48 @@
+export interface Workflow {
+  slug: WorkflowSlug;
+  name: string;
+  command: string;
+  tags?: [string];
+  description?: string;
+  arguments?: [Argument];
+  source_url?: string;
+  author?: string;
+  author_url?: string;
+  shells?: [Shell];
+}
+
+export interface Argument {
+  name: string;
+  description?: string;
+  default_value?: string;
+}
+
+export const enum Shell {
+  Bash,
+  Fish,
+  Zsh,
+}
+
+export type WorkflowSlug = string;
+
+function getFileNameOnly(filePath: string) : string {
+  return filePath.split("/").pop()!.split(".")!.shift()!;
+}
+
+const requireContext = require.context(
+  "yaml-loader!../specs",
+  false,
+  /\.ya?ml$/
+);
+
+export const WORKFLOWS = new Map<WorkflowSlug, Workflow>();
+
+requireContext.keys().forEach((key: string) => {
+  const obj = requireContext(key);
```

Comment:
I think something that is more idiomatic is:
```
  const fileNames = fs.readdirSync("../specs");

  fileNames.forEach((fileName) => {
        const slug = fileName.replace(/\.ya?ml$/, "");
```

Reply:
using the `fs` package is definitely the right approach if you want to generate the workflow objects at runtime, but the issue here is we want to generate the workflows at _build_ time instead. 

That means we have to use webpack and their [apis](https://webpack.js.org/guides/dependency-management/#requirecontext) (`requireContext.keys()..`)  to read the `specs` folder and generate the correct corresponding JS objects